tests/Resources/core.runtime.addontest.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+
+'use strict';
+
+var should = require('./utilities/assertions');
+
+describe('Core', () => {
+	describe('Runtime', () => {
+		describe('hasProperty', () => {
+			describe('Top-Module', () => {
+				describe('Submodules', () => {
+					it('should check for sub-module', () => {
+						Ti.should.have.property('UI');
+						// @fixme this check throws an error on android
+						if (Ti.Platform.osname !== 'android') {
+							Ti.should.not.have.property('Foo');
+						}
+					});
+				});
+
+				describe('Custom properties', () => {
+					it.ios('should check for custom properties', () => {
+						Ti.should.not.have.property('custom');
+						Ti.custom = {};
+						Ti.should.have.property('custom');
+					});
+				});
+			});
+
+			describe('Proxy', () => {
+				describe('Getter/Setter', () => {
+					it('should check for existing getter method', () => {
+						const tabGroup = Ti.UI.createTabGroup();
+						tabGroup.should.have.property('getTabs');
+					});
+
+					it.ios('should check for dynamic getter method', () => {
+						const view = Ti.UI.createView();
+						view.should.have.property('getFoo');
```

Comment:
why should this pass? It shouldn't respond true here as there is no "foo" property.

Reply:
That's iOS getter/setter magic. It first tries to find an actually existing property on the proxy and tries to set that. If not found, it will fallback to set it as a "dynamic property" on the proxy. So everything starting with `set` or `get` is automatically available on iOS.

https://github.com/appcelerator/titanium_mobile/blob/abc04066bd0ec9ac29066ba9a3b46fb489d51044/iphone/TitaniumKit/TitaniumKit/Sources/API/TiProxy.m#L1100

This results in these statements being equivalent on iOS:
```js
view.foo = 'bar;
console.log(view.foo);
```

```js
view.setFoo('bar);
console.log(view.getFoo());
```